src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -799,40 +799,44 @@ private static void checkInterfaceMethods(
     final List<String> missingMethodsNames = new ArrayList<String>();
 
     if (intReference.getHaxeClass() != null) {
+      List<HaxeMethod> methods = clazz.haxeClass.getHaxeMethods(null);
       for (HaxeMethodModel intMethod : intReference.getHaxeClass().getMethods(null)) {
         if (!intMethod.isStatic()) {
-          // Implemented method not necessarily located in current class
-          final PsiMethod[] methods = clazz.haxeClass.findMethodsByName(intMethod.getName(), true);
-          final PsiMethod psiMethod = ContainerUtil.find(methods, new Condition<PsiMethod>() {
-            @Override
-            public boolean value(PsiMethod method) {
-              return method instanceof HaxeMethod;
-            }
-          });
 
-          if (psiMethod == null) {
+          Optional<HaxeMethodModel> methodResult = methods.stream()
+            .filter(method -> intMethod.getName().equals(method.getName()))
+            .map(HaxeMethodPsiMixin::getModel)
+            .findFirst();
```

Comment:
If what we are trying to find is the implementation of something of an abstract function or an interface method, then this won't work reliably.  The problem is that getHaxeMethods() only gets the first encountered element of any given name, so if a method is declared in an interface, but implemented in a superclass (which may be dumb design, but works), then the interface can be found, and not the implementing method.  We need a version of getHaxeMethods which calls findNamedSubcompontents(unique:false,...), and then this will work as intended.

Reply:
i am not sure i understand the issue here, to me it looks like  `getHaxeMethods()` fetches every single method, its even mentioned in a comment, and `filterNamedComponentsByType` only filters on type making sure its all Methods (`HaxeComponentType.typeOf(component) == type;`)

i guess i could filter out anything that belongs  to an interface in my logic, but so far i have not seen that issue in my tests and debugging sessions. (if you got the time, a simple example on what would case a problem would be nice)
```
  @NotNull
  @Override
  public List<HaxeMethod> getHaxeMethods(@Nullable HaxeGenericResolver resolver) {
    // XXX: This implementation is equivalent to getAllMethods().  That
    //      may not be what we want.
    final List<HaxeNamedComponent> alltypes = HaxeResolveUtil.findNamedSubComponents(resolver, this);
    final List<HaxeNamedComponent> methods = HaxeResolveUtil.filterNamedComponentsByType(alltypes, HaxeComponentType.METHOD);
    final List<HaxeMethod> result = new ArrayList<>();
    for (HaxeNamedComponent method : methods) {
      result.add((HaxeMethod)method);
    }
    return result;
  }
```

Reply:
I guess i need to implement som kind of `getAllMethods` methond then, as the one that exists uses `PsiClassImplUtil` from  the Java plugin and does not really work  properly with  haxe.